sway/input/keyboard.c

```diff
@@ -317,16 +298,39 @@ static size_t keyboard_keysyms_translated(struct sway_keyboard *keyboard,
  * This will trigger keybinds such as Alt+Shift+2.
  */
 static size_t keyboard_keysyms_raw(struct sway_keyboard *keyboard,
-		xkb_keycode_t keycode, const xkb_keysym_t **keysyms,
+		xkb_keysym_t *keysyms,
 		uint32_t *modifiers) {
 	struct wlr_input_device *device =
 		keyboard->seat_device->input_device->wlr_device;
 	*modifiers = wlr_keyboard_get_modifiers(device->keyboard);
 
-	xkb_layout_index_t layout_index = xkb_state_key_get_layout(
-		device->keyboard->xkb_state, keycode);
-	return xkb_keymap_key_get_syms_by_level(device->keyboard->keymap,
-		keycode, layout_index, 0, keysyms);
+	size_t num_keysyms = 0, num_keysyms_for_keycode;
+	const xkb_keysym_t *keysyms_for_keycode;
+	xkb_keycode_t keycode;
```

Comment:
Style: these can move inside the loop. We generally only define outside the loop if we need the variables after the loop.